src/System.Net.Utilities/tests/FunctionalTests/PingTests.cs

```diff
@@ -58,7 +58,33 @@ public class PingTests
                 });
         }
 
-        [ActiveIssue(2487, PlatformID.AnyUnix)]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        [Fact]
+        public async Task SendPingAsyncWithIPAddressAndTimeoutAndBuffer_Unix()
+        {
+            byte[] buffer = TestSettings.PayloadAsBytes;
+            IPAddress localIpAddress = await TestSettings.GetLocalIPAddress();
+
+            SendBatchPingAsync(
+                (ping) => ping.SendPingAsync(localIpAddress, TestSettings.PingTimeout, buffer),
+                (pingReply) =>
+                {
+                    Assert.Equal(IPStatus.Success, pingReply.Status);
+                    Assert.True(pingReply.Address.Equals(localIpAddress));
+
+                    if (CanUseRawSockets())
+                    {
+                        Assert.Equal(buffer, pingReply.Buffer);
+                    }
+                    else
+                    {
+                        Assert.Equal(Array.Empty<byte>(), pingReply.Buffer);
+                    }
+                });
+        }
+
+        // Non-root pings cannot send arbitrary data in the buffer, and do not receive it back in the PingReply.
```

Comment:
At first I didn't get what you meant by non-root. 

Since this test is Windows-specific the comment can't really apply unless you are talking about a Windows privilege level.